synapseclient/client.py

```diff
@@ -2172,6 +2172,62 @@ def getTeamMembers(self, team):
         for result in self._GET_paginated('/teamMembers/{id}'.format(id=id_of(team))):
             yield TeamMember(**result)
 
+    def get_membership_status(self, userid, team):
+        """Retrieve a user's Team Membership Status bundle.
+        https://docs.synapse.org/rest/GET/team/id/member/principalId/membershipStatus.html
+
+        :param user: Synapse userid
+        :param team: A :py:class:`synapseclient.team.Team` object or a
+                     team's ID.
+
+        :returns: dict of TeamMembershipStatus"""
+        teamid = id_of(team)
+        request = "/team/{team}/member/{user}/membershipStatus".format(
+            team=teamid,
+            user=userid)
+        membership_status = self.restGET(request)
+        return membership_status
+
+    def invite_member_to_team(self, team, user=None, email=None,
+                              message=None):
+        """Invite user to a Synapse team via Synapse username or email
+        (choose one or the other)
+
+        :param syn: Synapse object
+        :param team: A :py:class:`synapseclient.team.Team` object or a
+                     team's ID.
+        :param user: Synapse username or profile id of user
+        :param email: Email of user
+        :param message: Additional message for the user getting invited to the
+                        team. Default to None.
+
+        Returns:
+            Invitation or None if user is already a member
+        """
+        if email is not None and user is not None:
+            raise ValueError("Must only specify 'user' or 'email'")
+
+        teamid = id_of(team)
+        is_member = False
+        # Message can be None
+        invite = {'teamId': str(teamid),
+                  'message': message}
+
+        if email is None:
+            userid = self.getUserProfile(user)['ownerId']
+            membership_status = self.get_membership_status(userid, teamid)
+            is_member = membership_status['isMember']
+            invite['inviteeId'] = str(userid)
+        else:
+            invite['inviteeEmail'] = email
+
+        if not is_member:
+            invite = self.restPOST("/membershipInvitation",
```

Comment:
make a post membership invitation function and then build up the body here and pass `**my_body` to the `POST` function.